pkg/webui/components/new-form/field/index.js

```diff
@@ -0,0 +1,210 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import bind from 'autobind-decorator'
+import classnames from 'classnames'
+import { getIn } from 'formik'
+
+import from from '../../../lib/from'
+import Icon from '../../icon'
+import Message from '../../../lib/components/message'
+import FormContext from '../context'
+import PropTypes from '../../../lib/prop-types'
+
+import style from './field.styl'
+
+export function getPassThroughProps (props, excludeProps) {
+  const rest = {}
+  for (const property of Object.keys(props)) {
+    if (!excludeProps[property]) {
+      rest[property] = props[property]
+    }
+  }
+  return rest
+}
+
+@bind
+class FormField extends React.Component {
+
+  static contextType = FormContext
+  static propTypes = {
+    className: PropTypes.string,
+    name: PropTypes.string.isRequired,
+    warning: PropTypes.message,
+    required: PropTypes.bool,
+    description: PropTypes.message,
+    onChange: PropTypes.func,
+    component: PropTypes.oneOfType([
+      PropTypes.func,
+      PropTypes.string,
+    ]).isRequired,
+  }
+
+  static defaultProps = {
+    onChange: () => null,
+    warning: '',
+    description: '',
+    required: false,
+  }
```

Comment:
Should we default `component` to `Input`, to stay in line with the old form's api?